authors/models.py

```diff
@@ -1,22 +1,61 @@
 import uuid
+from django.core.exceptions import ValidationError
 from django.db import models
+from django.urls import reverse 
 from django.contrib.auth.models import User
+from django.utils.translation import gettext_lazy as _
 
 # Create your models here.
 class Author(models.Model):
     id = models.UUIDField(primary_key=True, editable=False, default=uuid.uuid4)
     user = models.OneToOneField(User, on_delete=models.CASCADE) # one2one with django user
-    friends = models.ManyToManyField("self") # bidirectional/symmetrical by default
+    friends = models.ManyToManyField('Author', blank=True, symmetrical=True) # bidirectional/symmetrical by default, allow empty
 
     display_name = models.CharField(max_length=30) # maximum 30 chars for display name
-    url = models.URLField() # the url to the author profile
-    host = models.URLField() # the host server node url, ours is https://social-distance-api.herokuapp.com/
+    github_url = models.URLField(null=True) # the url to the author github profile
```

Comment:
When testing with the django admin I have to enter a value for this field, otherwise the author cannot be saved.

Looks like `null=True` does not make it actually optional, we will also need `blank=True`

The following is quoted from https://docs.djangoproject.com/en/3.2/ref/models/fields/#null:
```For both string-based and non-string-based fields, you will also need to set blank=True if you wish to permit empty values in forms, as the null parameter only affects database storage (see blank).```

Reply:
good catch. null and blank confused me.